zipline/protocol.py

```diff
@@ -88,8 +80,52 @@ def to_series(self, index=None):
         return pd.Series(self.__dict__, index=index)
 
 
+def _deprecated_getitem_method(name, attrs):
+    """Create a deprecated ``__getitem__`` method that tells users to use
+    getattr instead.
+
+    Parameters
+    ----------
+    name : str
+        The name of the object in the warning message.
+    attrs : iterable[str]
+        The set of allowed attributes.
+
+    Returns
+    -------
+    __getitem__ : callable[any, str]
+        The ``__getitem__`` method to put in the class dict.
+    """
+    attrs = frozenset(attrs)
+    msg = "'{0}[attr]' is deprecated, please use '{0}.attr' instead".format(
```

Comment:
I meant that we should include the actual attr that was requested (deferring the creation of the error string until `__getitem__` was called) so that the user can more easily find where they need to update their code.